bzzeth/bzzeth.go

```diff
@@ -266,10 +269,22 @@ func (b *BzzEth) deliverAndStoreAll(ctx context.Context, req *request, headers [
 
 	// wait for all validations to get over and close the channels
 	err := wg.Wait()
+
+	// Store all the valid header chunks in one shot
+	storeErr := b.storeChunks(ctx, chunks)
+	if storeErr != nil {
+		return storeErr
+	}
+
+	// Pass on the validation error if any
 	if err != nil {
 		return err
 	}
 
+	return nil
+}
+
+func (b *BzzEth) storeChunks(ctx context.Context, chunks []chunk.Chunk) error {
```

Comment:
iterating over results does not make sense if err is not nil. 
also doing the error comparison for every result is not necessary. 
I recommend just returning error if invalid, logging and returning nil if other error. 
otherwise log the number of new chunks

Reply:
I am now using the response bool[] array which indicates which chunks are stores and which are not.  SO it should be fine now.